### DISCUSSION TITLE
[ARCHIVED] Exercise 2 - 3D interactivity

### DISCUSSION THREAD

**Discussion by periode** [periode/augmenting-gallery#3] [Homework]
- Make at least 2 GameObjects appear and disappear through scripts (using `gameObject.SetActive()`).
- Thinking of [Augmented Exercise 1](https://github.com/periode/augmenting-gallery/wiki/Augmented-Exercises#exercise-1---object-story), import a 3D model of an object into your Unity scene, along with images which can enrich our understanding of that object.
- Find 3D models websites on the [Resources](https://github.com/periode/augmenting-gallery/wiki/Resources#3d-models) page.
- Find more info on importing 3D models on the [Unity Manual](https://docs.unity3d.com/Manual/ImportingModelFiles.html)

Post a screenshot of your work with a couple of sentences describing, for instance, what you did, how you did it, and why you did it.

**Comment by yylinj** [comment]
here's the screen recording of my objects appearing and disappearing through pressing keys
[](url)
https://user-images.githubusercontent.com/98582179/153073938-058539cc-b794-441a-80ef-d4350ed1fa76.mov

here's a screenshot of the codes (I think I made a mistake in there because the appearing and disappearing is not as functioning as I wished 😭 I have to first select the object)
<img width="974" alt="Screen Shot 2022-02-08 at 9 42 50 PM" src="https://user-images.githubusercontent.com/98582179/153074028-2155c868-97c2-44c1-9644-ea9159d36b06.png">

I tried to download a 3D statue of the Goddess of Hope from Sketchfab ("Goddess of hope" (https://skfb.ly/VMo9) by Geoffrey Marchal is licensed under Creative Commons Attribution (http://creativecommons.org/licenses/by/4.0/).), but the download file is a folder and I cannot just drag the obj into the project (for some reason when I drag it in there, it shows no object but just xyz axis). I'll post this post first, and if I can find a way from google on how to import it, I will post another post on the screenshot of the imported object.

**Comment by mdl601** [comment]
![A1FBAEBA-FBB1-41FE-9AED-9D06FAE19777](https://user-images.githubusercontent.com/98850518/153078832-eb4dab6d-b233-403c-9e73-e038425c3f19.jpeg)

I think I did something wrong here- my object disappears, but doesn't reappear the way it's supposed to. I've double checked my code and I'm not entirely sure what's wrong with it, everything looks good with the code (I think)

**Comment by aidanmassie** [comment]
See below a screenshot of my Unity project and the code. I'm not getting any error messages within the Unity console, but the arrow keys aren't making either object disappear or appear. I did, however, get the two 3D models imported into the project--don't make fun, just a casual Star Wars fan here. I included an image of the pilot of each respective plane, as well as the Star Wars logo as some help for those who couldn't (or don't care enough to) put it all together.

<img width="1437" alt="Screen Shot 2022-02-08 at 4 33 33 PM" src="https://user-images.githubusercontent.com/60477421/153079604-7eb49914-15e9-4c4f-9484-5d41f91be480.png">
<img width="1429" alt="Screen Shot 2022-02-08 at 4 33 47 PM" src="https://user-images.githubusercontent.com/60477421/153079619-ca34aaf4-9ca5-4034-9eee-4274c6ce8a7a.png">

**Comment by miafan716** [comment]
Here is what I have for this week:

https://user-images.githubusercontent.com/55967980/153080001-5fda8202-5607-4959-b33f-6b63c7ab4301.mp4

I tried to describe a scene that Kirby flying through these Nintendo Characters, fighting back them and finally getting into the arcadia gallery painting. The interactions are pressing "S" to let squirtle appear and disappear, "B" for bulbasaur in similar way. I mainly followed the instruction in lecture for realizing the interaction. 
<img width="960" alt="AtG W3 Mia Code" src="https://user-images.githubusercontent.com/55967980/153080035-89188159-2770-4447-903f-f7bb6f4b89ca.png">

Some problems are: 
I downloaded all the 3d models from sketchfab. Some of them were obj. and one of them was fbx., but not all of them have embedded materials and I don't know how to get rid of this grey default material. I can add some by myself but the lighting effects would be too ugly.

**Comment by joshblecherman** [comment]
<img width="830" alt="Screen Shot 2022-02-09 at 12 08 07 AM" src="https://user-images.githubusercontent.com/57162769/153091150-5a72b6ee-55e4-45c3-aa34-4b88df9022c0.png">

My scene is a lewis chessman with four pictures in the background. There is a walrus because the piece was carved from a walrus tooth, a modern chess king, the place it was discovered, and the current chess world champion Magnus Carlsen. I have not added any interactivity yet as I had some trouble downloading xcode.

**Comment by palomabrites** [comment]
<img width="1440" alt="screenshot1" src="https://user-images.githubusercontent.com/98867529/153091274-c0fb369f-2521-43d2-8ce2-3e01cb1ccffa.png">
<img width="1440" alt="sc2" src="https://user-images.githubusercontent.com/98867529/153091294-df2c4e45-4bf6-40e1-ba25-f29bcc60a73a.png">
I downloaded a cell model from sketch lab. I made this with the intention of revealing the images underneath however the code only works for one image at a time and if I try to use the other key then both of them stop working.

**Comment by chloeumlaut** [comment]
<img width="377" alt="Screen Shot 2022-02-09 at 9 31 31 AM" src="https://user-images.githubusercontent.com/98907766/153156304-a06e2272-5363-4d1f-9546-01c0d6d54b26.png">

I'm not exactly sure what I was going for with this, it took me quite some time to get the code right. I started by adding two spheres and the button, then I used the gameObject script to make the spheres disappear when the button was pressed. I saw this free little chicken guy on SketchFab and thought, hey! he can "eat" those spheres! So here he is eating one of the spheres.

**Comment by vb1424** [comment]
Here is a screen recording of my objects disappearing by pressing the space bar. I made a tree house, so when they disappear (the planes that make up the tree house), you are left with a simple essence of nature :). Also a screenshot of my code

<img width="640" alt="Screen Shot 2022-02-15 at 12 26 36 AM" src="https://user-images.githubusercontent.com/98861299/153964002-f96b1ca1-7732-479f-944e-91d0a5ca67de.png">

https://user-images.githubusercontent.com/98861299/153965011-0d6e8cd5-865c-409f-ad1d-dcf0a4e97f94.mov

.

**Comment by fractalessense** [comment]
https://user-images.githubusercontent.com/98880710/154045808-b129bf1c-7e53-4481-911f-052c454933be.mov

I used a 3D model of Kokura castle. The images reveal more perspectives on the building as it fits into the modern landscape.

**Comment by luyumenq** [comment]
https://user-images.githubusercontent.com/72938590/190922697-138842a7-a5d3-4a71-bf83-06ea5e9f769c.mp4

<img width="997" alt="Screenshot 2022-09-18 at 20 32 11" src="https://user-images.githubusercontent.com/72938590/190922878-0a0ce11b-ceeb-4640-acc8-bbdce1644667.png">

<img width="456" alt="Screenshot 2022-09-18 at 20 30 33" src="https://user-images.githubusercontent.com/72938590/190922825-d4cd3240-d0fd-4da1-9126-bb61ad0c6fa0.png">

 I am expanding the expressiveness of my virtual gallery by adding the model ship-in-a-bottle and I added this switch on the top-left corner of the screen to switch things on and off. I have to write a script to ensure the button is not a one-way process. However, when I was playing with the parameters from my asset of downloaded model, I find it hard to exactly reenact its preview on the webpage.

**Comment by kenparks12** [comment]
<img width="1111" alt="Screen Shot 2022-09-20 at 19 35 38" src="https://user-images.githubusercontent.com/97991944/191326582-5b14a35c-3f92-4065-a08e-4d1af6863b00.png">
<img width="1123" alt="Screen Shot 2022-09-20 at 19 35 03" src="https://user-images.githubusercontent.com/97991944/191326625-a3cbf83e-9e4a-4e96-994e-bc9c10a6d8a5.png">
<img width="1120" alt="Screen Shot 2022-09-20 at 19 34 50" src="https://user-images.githubusercontent.com/97991944/191326627-73bd43bc-5215-4f41-a2d3-8e915524fa58.png">

**Reply by kenparks12** [reply]
for my project, I got a 3D model of a bedroom from the Unity Assets store and created a canvas button that added posters to the wall as well as give information about the function of the tv. You can also press the second button to make the posters disappear again.

**Comment by dovebardin** [comment]
<img width="813" alt="Screen Shot 2022-09-20 at 10 45 25 PM" src="https://user-images.githubusercontent.com/113513286/191362333-11a3740e-b8cb-42b5-b7d5-4b7a1c03aa27.png">
<img width="810" alt="Screen Shot 2022-09-20 at 10 46 40 PM" src="https://user-images.githubusercontent.com/113513286/191362359-8c4a05f6-4186-4368-9de9-38701a6bbe96.png">

**Comment by mariaabzm** [comment]
Was having a lot of difficulties....  💔
<img width="679" alt="Screen Shot 2022-09-20 at 11 00 51 PM" src="https://user-images.githubusercontent.com/113439496/191363917-25f6298e-8592-4c83-8140-b9e82b021bee.png">
<img width="641" alt="Screen Shot 2022-09-20 at 11 01 15 PM" src="https://user-images.githubusercontent.com/113439496/191363924-faee4354-fb07-4074-b330-22867f2c7e8c.png">

**Comment by kylewenchang** [comment]
<img width="992" alt="Screen Shot 2022-09-20 at 11 15 18 PM" src="https://user-images.githubusercontent.com/63214975/191366475-8af3bb37-9b91-406a-8029-e1816f4e0783.png">
<img width="987" alt="Screen Shot 2022-09-20 at 11 18 16 PM" src="https://user-images.githubusercontent.com/63214975/191366510-235eb42e-c149-4e4f-821b-b1dee63b5522.png">
<img width="987" alt="Screen Shot 2022-09-20 at 11 17 19 PM" src="https://user-images.githubusercontent.com/63214975/191366538-e563c2a9-0e20-4075-9a70-72fbe2185a52.png">


https://user-images.githubusercontent.com/63214975/191367513-f7ce6e01-3937-4f40-b006-bf17f4f6040a.mov



For the prompt, I was thinking about something that disappears and reappears and I thought of a disappearing bunny magic trick. I downloaded a bunny and hat model from a 3D printing models website, fixed some stuff in Blender for import, and gave them materials in Unity. I have 2 buttons, a disappear button which hides the bunny, the "excited text," and the particles, and shows the normal text, and a reappear button, which hides the normal text to show the excited text, shows the bunny, and shows a puff of smoke particles.

**Comment by olivestew** [comment]
<img width="658" alt="Screenshot 2022-09-21 at 02 13 37" src="https://user-images.githubusercontent.com/91790610/191386694-7d84cd5b-cf7e-41dc-8b78-5f61884a74e0.png">

**Comment by izzyrieken** [comment]
![Uploading Screen Shot 2022-09-21 at 8.27.39 AM.png…]()
<img width="762" alt="Screen Shot 2022-09-21 at 8 27 29 AM" src="https://user-images.githubusercontent.com/58444126/191430207-cd1b7023-0543-469c-be5c-79bd9f4c0f4b.png">

**Comment by Dav1dyang** [comment]
This is a Pixar room! Click the word to see what happens! (Beach ball falling off from the ceiling)
<img width="1002" alt="Screen Shot 2022-09-21 at 8 33 47 AM" src="https://user-images.githubusercontent.com/55843005/191431262-1b4d9e80-a9a5-461a-9181-df240020df0c.png">

<img width="1004" alt="Screen Shot 2022-09-21 at 8 55 40 AM" src="https://user-images.githubusercontent.com/55843005/191435681-e362a8bb-7bd1-4b63-bf94-f8a1f7ef7bfc.png">



Here's a link to the video recording: https://youtu.be/GQ5jLUer2zQ

**Comment by vasantsun29** [comment]
<img width="497" alt="Screen Shot 2022-09-21 at 8 39 33 AM" src="https://user-images.githubusercontent.com/113478467/191432684-76d52366-23ae-4916-a373-baf0bc6d23bc.png">

<img width="431" alt="Screen Shot 2022-09-21 at 8 39 42 AM" src="https://user-images.githubusercontent.com/113478467/191432692-53fe5e1b-802b-405d-be5b-9dcdad8e0850.png">

Here I have a button that disappears a wall and reappears the museum guide who will show you around the memeseum.

**Comment by IanPort** [comment]
<img width="674" alt="ButtonInstall1" src="https://user-images.githubusercontent.com/113472376/191609365-04fcdc25-f820-4a3a-86df-e3f756b96d98.png">
<img width="648" alt="ButtonInstallation" src="https://user-images.githubusercontent.com/113472376/191609379-a098882f-c75e-4806-a601-725acdbaf404.png">

In building this viewing/gallery room, I thought about the display case as a practical and emotional tool. On the pedestal is an abstract self-portrait with 3D models, which becomes contained in glass and put under three spotlights after the button is pressed.

**Comment by Yupu-Chen** [comment]
https://user-images.githubusercontent.com/112383893/216845214-9c215100-1975-4343-988d-ae6b8dec7a88.mov

I use two buttons to make two objects appear and disappear respectively. And this is the script I attached to the two buttons.

<img width="1075" alt="Script" src="https://user-images.githubusercontent.com/112383893/216845284-531770c8-83f7-40d6-8190-1354f40ab0f2.png">

For the project, I downloaded a Boeing 787 model from the internet. I used images of the two kinds engines B-787 uses and briefly introduced them. I chose this object because airplanes are a significant milestone in human history. Ever since the invention of the first model of aircrafts by Wright Brothers, people have been working hard to increase their mobilities. Nowadays, people could easily travel to another continent within one day. However, along with such easiness of travel come many issues related to globalization and inequality. I'd like to explore more about this topic with such an object.

<img width="1651" alt="Screenshot 2023-02-05 at 9 42 52 PM" src="https://user-images.githubusercontent.com/112383893/217479766-aaa40952-9a5e-4ab0-8b3f-1676790aab1b.png">

**Comment by Opal42** [comment]
https://user-images.githubusercontent.com/123597298/216876435-aa866917-e838-4ce2-b4cb-be2e6e6df86b.mp4

I built a camera-tied HUD (forgot to show movement in the recording, but works as intended), that can pull up information within the HUD itself, as well as toggle different lighting options for the piece. Need to go do more research to find more information about this piece for the final assignment though. Also thinking about whether it would be best to have one button to toggle the other buttons, something that allows for a seamless view of the object itself. Open to suggestions.

**Comment by mariamabdelrazekk** [comment]
Noticing that the subway is a place where a lot of people experience different genres of music at the same time, I tried to represent that by using a 3D subway station with a couple of posters representing different genres of music that people listen to. I found it really hard to control the dimensions of the images that I uploaded into walls of the subway station. I just used three cubes and edited them to place the images on top of them. 

<img width="1440" alt="Screen Shot 2023-02-07 at 11 20 55 PM" src="https://user-images.githubusercontent.com/91086698/217384756-a59f2964-2046-4ee8-b625-8159bc6d77a1.png">
<img width="1440" alt="Screen Shot 2023-02-07 at 11 20 32 PM" src="https://user-images.githubusercontent.com/91086698/217384757-ac1f537c-023c-427f-b3c0-a9f147e2c072.png">

**Comment by aacarcrash** [comment]
[YouTube Link to a Demo of the Gallery](https://www.youtube.com/watch?v=7ZAtKHa-ofQ)

The project aims to facilitate a conversation regarding Orientalism, especially in the context of the British colonialism of India. The sculpture chosen is present in the The British Museum, London and was acquired in India in 1921 (a time period when India was still colonised).

**Comment by cz2109** [comment]
![image](https://user-images.githubusercontent.com/123963318/217410917-071cafe5-94da-41ab-b26f-2fc267ca6913.png)
An exhibition of the sculpture _Ain Sakhri lovers_. The piece surrounds two lovers from the Natufian period in an ancient rendering of human sexuality. There is a sense of timelessness in the piece, especially with the lack of faces. Surrounding the piece are photographs of the Levant area on one side, and queer movements on the other (Haring's work on the AIDs crisis and a photograph of the 1969 Stonewall Riots). Queerness, specifically notions of homoeroticism, permeates the piece with the absence of gender and its indubitably phallic imagery.

**Comment by hannah-y-zhao** [comment]
[Video link](https://youtu.be/wv8qsIJ1m0k)
I got the chest from the asset store, and just wanted to know about the history of chests

**Comment by daniaezz** [comment]
<img width="1440" alt="Screen Shot 2023-02-08 at 7 46 10 AM" src="https://user-images.githubusercontent.com/90758857/217454857-f54efd70-655c-468f-bd84-d75c23fbb4c8.png">

Bell peppers are often used as a reference for beginner artist or as a starting point to learning how to sketch. There is also a famous photograph of a bell pepper and I thought it would be interesting to show the different artistic aspects of an object as simple as a bell pepper

**Comment by emersonmack** [comment]
![Screen Shot 2023-02-13 at 5 57 56 PM](https://user-images.githubusercontent.com/123585567/218522581-00194dbb-44e8-49fd-b38e-6a6c7d2bd0b5.png)
![Screen Shot 2023-02-13 at 5 57 26 PM](https://user-images.githubusercontent.com/123585567/218522597-59960131-4bef-4158-91e9-204bd45d9882.png)
![Screen Shot 2023-02-13 at 5 55 13 PM](https://user-images.githubusercontent.com/123585567/218522606-4ccd9b20-2d92-4849-9a24-52e5d6f8f9df.png)
I found this lion and really liked it. I currated this gallery to show information on the left side (because you read left to right) and the lion front and center. The floor mimics the red grante of the statue to integrate the piece more with the museum environment.
